proxy/src/conditional.rs

```diff
@@ -0,0 +1,56 @@
+use std;
+
+/// Like `std::option::Option<C>` but `None` carries a reason why the value
+/// isn't available.
+#[derive(Clone, Debug)]
+pub enum Conditional<C, R> where
+    C: Clone + std::fmt::Debug,
+    R: Clone + std::fmt::Debug
```

Comment:
It shouldn't block this review, but the formatting used here is used nowhere else in the project as far as i can tell. We already have a smattering of styles, and I'd prefer not to introduce an N+1th form.  Perhaps just run rustfmt over the file before merging?

Reply:
I did the rustfmt.